src/host/outputStream.cpp

```diff
@@ -513,6 +513,19 @@ bool ConhostInternalGetSet::PrivateEraseAll()
     return SUCCEEDED(DoSrvPrivateEraseAll(_io.GetActiveOutputBuffer()));
 }
 
+// Method Description:
+// - Retrieves the current user default cursor style.
+// Arguments:
+// - style - Structure to receive cursor style.
+// Return Value:
+// - true if successful. false otherwise.
+bool ConhostInternalGetSet::GetUserDefaultCursorStyle(CursorType& style)
+{
+    const auto& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
+    style = gci.GetCursorType();

```

Comment:
If we’re on older Windows 10 or Windows 7, will this still work? I assume it will always return legacy.

Reply:
I believe the VT support was only added to conhost in build 10586 of Windows 10, so nothing prior to that should have any of this code. Maybe one of the core devs can confirm this, but I don't think we need to worry about older versions of Windows at all.

Reply:
Yeah, this is totally fine. Conhost is also shipped as a totally independent unit -- so if conhost can run on an older version of Windows, it will bring with it **full VT support** and this cursor change.

(incidentally, it works on Windows 8.1 and brings full VT support :wink:)